apps/studio/src/pages/sites/[siteId]/admin.tsx

```diff
@@ -0,0 +1,228 @@
+        message: "Site theme must be present",
+        path: ["theme"],
+      })
+      .refine((data) => !!data.navbar, {
+        message: "Site navbar must be present",
+        path: ["navbar"],
+      })
+      .refine((data) => !!data.footer, {
+        message: "Site footer must be present",
+        path: ["footer"],
+      }),
+    defaultValues: {
+      config: JSON.stringify(previousConfig, null, 2),
+      theme: JSON.stringify(previousTheme, null, 2),
+      navbar: JSON.stringify(previousNavbar.content, null, 2),
+      footer: JSON.stringify(previousFooter.content, null, 2),
+    },
+  })
+
+  const [nextUrl, setNextUrl] = useState("")
+  const isOpen = !!nextUrl
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      if (isDirty) {
+        router.events.off("routeChangeStart", handleRouteChange)
+        setNextUrl(url)
+        router.events.emit("routeChangeError")
+        // eslint-disable-next-line @typescript-eslint/only-throw-error
+        throw "Error to abort router route change. Ignore this!"
+      }
+    }
+
+    if (!isOpen) {
+      router.events.on("routeChangeStart", handleRouteChange)
+    }
+    return () => {
+      router.events.off("routeChangeStart", handleRouteChange)
+    }
+  }, [isOpen, router.events, isDirty])
+
+  const onClickUpdate = handleSubmit((input) => {
+    mutate({
+      siteId,
+      ...input,
+    })
+  })
+
+  if (!isUserIsomerAdmin) {
+    return (
+      <Center h="full">
+        <Text>You do not have permission to access this page.</Text>
+      </Center>
+    )
+  }
+
+  return (
+    <>
+      <UnsavedSettingModal
+        isOpen={isOpen}
+        onClose={() => setNextUrl("")}
+        nextUrl={nextUrl}
+      />
+      <chakra.form
+        onSubmit={onClickUpdate}
+        overflow="auto"
+        height={0}
+        minH="100%"
+      >
+        <Center py="5.5rem" px="2rem">
+          <VStack w="48rem" alignItems="flex-start" spacing="1.5rem">
+            <Text w="full" textStyle="h3-semibold">
+              Manage site configurations
+            </Text>
+            <Infobox variant="warning" textStyle="body-2" size="sm">
+              No validation is done on the JSON input. Please ensure that they
+              are valid before saving.
+            </Infobox>
+
+            {SUPPORTED_SITE_CONFIG_TYPES.map((type) => (
+              <FormControl key={type} isInvalid={!!errors[type]}>
+                <VStack w="full" alignItems="flex-start" spacing="0.75rem">
+                  <Text
+                    textColor="base.content.strong"
+                    textStyle="subhead-1"
+                    pt="0.5rem"
+                  >
+                    Site {type}
+                  </Text>
+
+                  <Textarea
+                    fontFamily="monospace"
+                    boxSizing="border-box"
+                    minH="18rem"
+                    {...register(type)}
+                  />
```

Comment:
not actionable: see if it's easy to format - i found [this](https://github.com/microlinkhq/react-json-view) which seems to be the one used by our `play.isomer.gov.sg`

Reply:
Hmm I didn't intend to do formatting here because main workflow from migrators is that they will edit on VSCode first before pasting in here.